### PR TITLE
feat (IAM Access Groups): Add support for access group templates

### PR DESCRIPTION
**Pull request by hariarla**
## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

SDK adopters will be able to call access group template and assignment APIs.
## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

## Other information

API Definition
[Stage](https://test.cloud.ibm.com/docs/access-groups)
[Production](https://cloud.ibm.com/docs/access-groups)

Integration Tests:
Screenshot 2023-07-19 at 8 57 02 PM

Example Tests
Screenshot 2023-07-19 at 9 08 20 PM

### CODE REVIEW

**Review by padamstx — CHANGES_REQUESTED** [review]
Looks good overall... just one minor change to use the production service URL

File: ibm_platform_services/iam_access_groups_v2.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 class IamAccessGroupsV2(BaseService):
     """The iam-access-groups V2 service."""
 
-    DEFAULT_SERVICE_URL = 'https://iam.cloud.ibm.com'
+    DEFAULT_SERVICE_URL = 'https://iam.test.cloud.ibm.com'
```
[the remark this answers is not in the document]
Comment (hariarla) [reply ibm_platform_services/iam_access_groups_v2.py]: fixed it.

**Review by padamstx — APPROVED** [review]
LGTM

File: examples/test_iam_access_groups_v2_examples.py [hunk 2]
```diff
@@ -511,6 +522,587 @@ def test_update_account_settings_example(self):
...
+            pytest.fail(str(e))
+
+    @needscredentials
+    def test_get_template_version_example_2(self):
+        """
+        get_template_version request example 2
+        """
+        try:
+            print('\nget_template_version() result 2:')
+            # begin-get_template_version
```
Comment (padamstx) [inline examples/test_iam_access_groups_v2_examples.py:910]: Just noticed that you also have duplicate examples for this operation in this PR (just reviewed the Go PR and noticed the same).

**Review by padamstx — CHANGES_REQUESTED** [review]
When I initially reviewed this PR, I missed the duplicate code examples for the get_template_version operation.  You can have only 1 code example per operation.

**Comment by hariarla** [comment]
**Unit Tests:**
Screenshot 2023-07-25 at 12 02 24 AM

**Integration Tests:**
Screenshot 2023-07-25 at 12 06 53 AM

**Example Tests:**
Screenshot 2023-07-25 at 12 31 18 AM

File: examples/test_iam_access_groups_v2_examples.py (hunk 2 above)
Comment (hariarla) [reply examples/test_iam_access_groups_v2_examples.py]: Removed duplicate example tests. will do the same for Go and Node aswell.

**Review by pyrooka — CHANGES_REQUESTED** [review]
Almost good, please fix that one small issue.

File: examples/test_iam_access_groups_v2_examples.py [hunk 3]
```diff
@@ -511,6 +522,539 @@ def test_update_account_settings_example(self):
...
+                name='IAM Admin Group template',
+                account_id=test_account_id,
+                description='This access group template allows admin access to all IAM platform services in the account.',
+                group=access_group_input_model,
+                policy_template_references=[policy_templates_input_model],
+            )
+            create_template_response = response.get_result()
+
+            print(json.dumps(create_template_response, indent=2))
+            # end-create_template
```
[the remark this answers is not in the document]
Comment (pyrooka) [reply examples/test_iam_access_groups_v2_examples.py:604]: A new line is missing before the `end` tag.

**Review by padamstx — APPROVED** [review]
LGTM

**Comment by hariarla** [comment]
Minor changes to schemas.
Tests:
Screenshot 2023-08-20 at 11 07 40 PM
Screenshot 2023-08-20 at 11 01 54 PM
Screenshot 2023-08-18 at 7 26 03 PM

**Pull request merged**